test/vcf/optional_policy_test.cpp

```diff
@@ -137,4 +137,215 @@ namespace ebi
+                                123456,
+                                { "id123" },
+                                "A",
+                                { "AC", vcf::GVCF },
+                                1.0,
+                                { vcf::PASS },
+                                { { vcf::END, "0" } },
+                                { vcf::GT, "XYZ" },
+                                { "0", "1:9" },
+                                sources[0]})) );
+
+            CHECK_NOTHROW( (optional_policy.optional_check_body_entry(parsing_state3, vcf::Record{
+                                1,
+                                "chr1",
+                                123456,
+                                { "id123" },
+                                "A",
+                                { vcf::GVCF },
+                                1.0,
+                                { vcf::PASS },
+                                { { vcf::END, "0" } },
+                                { vcf::GT, "XYZ" },
+                                { "0/0/0:12", "1|0|1:5" },
+                                sources[2]})) );
+
+            CHECK_NOTHROW( (optional_policy.optional_check_body_entry(parsing_state2, vcf::Record{
+                                1,
+                                "chr1",
+                                123456,
+                                { "id123" },
+                                "A",
+                                { vcf::GVCF },
+                                1.0,
+                                { vcf::PASS },
+                                { { vcf::END, "0" } },
+                                { vcf::GT },
+                                { "0/0" },
+                                sources[1]})) );
+
+            CHECK_NOTHROW( (optional_policy.optional_check_body_entry(parsing_state2, vcf::Record{
+                                1,
+                                "chr1",
+                                123456,
+                                { "id123" },
+                                "A",
+                                { "AC", vcf::GVCF },
+                                1.0,
+                                { vcf::PASS },
+                                { { vcf::END, "0" } },
+                                { "XYZ" },
+                                { "15" },
+                                sources[1]})) );
+
+            CHECK_THROWS_AS( (optional_policy.optional_check_body_entry(parsing_state2, vcf::Record{
+                                1,
+                                "chr1",
+                                123456,
+                                { "id123" },
+                                "A",
+                                { vcf::GVCF },
+                                1.0,
+                                { vcf::PASS },
+                                { { vcf::END, "0" } },
+                                { vcf::GT },
+                                { "0|1" },
```

Comment:
this reminds me, is there any `CHECK_NO_THROW( ... "0|0"...`?

Reply:
Shouldn't the sole presence of `0|0` generate a warning? 
also, is `0/0` the same as `0|0`?

Reply:
OK, we never really explained a lot of the biological background about VCFs, the vertical line means that the genotypes are phased, which is that it is known to which chromosome the allele belongs. you know that diploid species have 2 copies of each chromosome (humans have 23 pairs) so the genotype reports the allele found in the same position of both copies of a chromosome. So `0/1` is the same as `1/0`, because the `/` means unphased (it's not known which number corresponds to each copy), but `0|1` is **not** the same as `1|0`. In summary, an appearance of `0|0` should prevent raising this warning

Reply:
oh ok thanks